sdk/servicebus/service-bus/test/operationOptionsForATOM.spec.ts

```diff
@@ -0,0 +1,262 @@
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.createSubscription(
+            { topicName: enitityName1, subscriptionName: enitityName2 },
+            {
+              abortSignal: AbortController.timeout(1)
+            }
+          )
+      );
+    });
+    it("getSubscription", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.getSubscription(enitityName1, enitityName2, {
+            abortSignal: AbortController.timeout(1)
+          })
+      );
+    });
+    it("updateSubscription", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.updateSubscription(
+            { topicName: enitityName1, subscriptionName: enitityName2 },
+            {
+              abortSignal: AbortController.timeout(1)
+            }
+          )
+      );
+    });
+    it("deleteSubscription", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.deleteSubscription(enitityName1, enitityName2, {
+            abortSignal: AbortController.timeout(1)
+          })
+      );
+    });
+    it("getSubscriptionRuntimeinfo", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.getSubscriptionRuntimeInfo(
+            enitityName1,
+            enitityName2,
+            {
+              abortSignal: AbortController.timeout(1)
+            }
+          )
+      );
+    });
+    it("getSubscriptions", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.getSubscriptions(enitityName1, {
+            abortSignal: AbortController.timeout(1)
+          })
+      );
+    });
+    it("getSubscriptionsRuntimeinfo", async () => {
+      await verifyAbortError(
+        async () =>
+          await serviceBusAtomManagementClient.getSubscriptionsRuntimeInfo(enitityName1, {
+            abortSignal: AbortController.timeout(1)
+          })
+      );
+    });
+  });
+
+  describe("RequestOptions timeout", () => {
+    // This test makes sure the timeout is plumbed as expected
+    it("requestOptions.timeout should work", async () => {
+      try {
+        await serviceBusAtomManagementClient.createQueue(enitityName1, {
+          requestOptions: { timeout: 1 }
+        });
+        assert.fail();
+      } catch (err) {
+        assert.equal(err.name, "AbortError");
+        assert.equal(err.message, "The operation was aborted.", "Unexpected error caught: " + err);
+      }
+    });
+  });
+
+  describe.skip("RequestOptions custom headers", () => {
+    // Response doesn't contain the custom headers,
+    // Custom headers are being plumbed properly at the SDK level, might be a problem at the core-http or at the service
+    // Needs more investigation
+    it("requestOptions.customHeaders should work", async () => {
```

Comment:
customHeaders aren't being present in the response.

Reply:
I think custom headers are only applied to the request. I don't think the service has to respect them or return them in a response.
You'd have to spy on the client's `sendRequest` method and inspect the `httpRequest` that was passed to it in order to see if the headers are set (though really I think that should probably be a test in core-http.) I'm fine if we don't have a test explicitly testing this in service-bus unless we're not using core-http the same way other clients do, in which case spying on `sendRequest` should suffice.

Reply:
Updated the test.